pkg/engine/internal/planner/logical/planner.go

```diff
@@ -144,6 +164,9 @@ func buildPlanForLogQuery(
 		},
 	)
 
+	// Do we need a projection of all comlumns right after maketable?
+	// builder = builder.ProjectAll(false, false)
```

Comment:
we could, but i expect optimiser to remove it anyways after pushing it down the Scan node.

I like the idea though of adding Projection explicitly to the logical plan. For example, the plan could explicitly call out `RangeAggregation` needs `timestamp, group_by` columns using a projection node placed right before it.

Should make it easier to debug as these things currently happen implicitly via the optimiser.